lib/hls/hls_parser.js

```diff
@@ -2587,6 +2588,9 @@ shaka.hls.HlsParser = class {
           this.manifest_.startTime = this.startTime_;
         }
       } catch (e) {
+        if (e.code === shaka.util.Error.Code.OPERATION_ABORTED) {
+          return;
+        }
         stream.closeSegmentIndex();
```

Comment:
Why would you not close the segment index after aborting here?  Could this leak segment references?

Reply:
Theoretically abort is triggered by closeSegmentIndex implementation, that was my main reason for omitting that. Being catious wouldn't harm though, so I've added this call back.